### PR TITLE
Clean up RcrainfoClient API and remove new client

### LINKED ISSUE #1530: Update Docs before 3.0 release
Title says it all

The emanifest 3.0 docs should cover these new features (if implemented, decision pending on some)

- [x] Basic Usage
- [x] mention how to obtain API credentials
- [x] auto-authentication
- [x] Regulator flag (pending discussion)
- [x] Encoding and uploading multipart/mixed attachments
- [x] Methods intended to be overridden
- [x] requests.Session API
- [x] Changes from emanifest@2.X
    * The 3.0 intends to be stable, sorry for the complete rewrite necessary to update from 2.x

### LINKED ISSUE #1531: Update or remove `new_client`
Right now the `new_client` function only initiates a new RcrainfoClient with a base_url but has no way of providing api credentials or any of the other __inti__ params.

We can either...
1. get rid of it
2. modify it to accept the same parameters as the RcrainfoClient __init__ method
    * This doesn't really add any value, if it just does this, I would prefer just remove it
3. modify it so it provides some business value. 
    * Here are some of things I can think of, but open to something else. I'm not sure this is enough to keep it around. 
    * initiate the RcrainfoClient while reading some values from the environment.

This is certainly an opinionated approach, I don't see it adding that much value (enough worth the 'liability' so to speak)

### PR DESCRIPTION
**Pull request by dpgraham4401**
This PR marks the final installment of our emanifest 3.0 improvement series. 

It cleans up the API (open to opinions and modifications) via getters and setters (@properties), removes the `new_client` function, and updates the docs (README.md). 

After this, we should be ready (or getting close barring any additional backwards incompatible changes we'd like to make before release) to release version 3.0.

The Publish to PyPI workflow should only [run on merged PR to master when pyproject.toml has changed](https://github.com/orgs/community/discussions/26724) (assuming our master branch has some branch protection rules, should be OK even without)

closes #1530 
closes #1531

### CODE REVIEW

**Comment by dpgraham4401** [comment]
@scottdchristian @wiljnichepa 

I'd like to keep the emanifest PyPI package API pretty stable after 3.0 release (Sorry users of the 2.0), y'all have any ideas you'd like to see implemented in this package before we release 3.0?

**Comment by wiljnichepa** [comment]
I think this looks great - I will review in depth this week and have full comments by Friday.

**Comment by scottdchristian** [comment]
Yes I will plan on looking too.  Thanks for asking!

File: emanifest-py/README.md [hunk 1]
```diff
@@ -14,140 +14,218 @@ refactoring.
...
 
 Before using the **emanifest** package, ensure you have a RCRAInfo user account and
 the [necessary permissions](https://www.epa.gov/e-manifest/frequent-questions-about-e-manifest#user_question6) to
 generate an API ID and key.
 
-All methods to access the e-Manifest APIs are implemented by the `RcrainfoClient` class which needs to be authenticated
-with your API ID and Key. A new instance of the class can be initiated with the `new_client()` convenience function
-like so:
+All methods to access the e-Manifest APIs are implemented by the `RcrainfoClient` class which needs your API ID and Key
+to authenticate . A new instance of the class can be initiated like so:
```
Comment (wiljnichepa) [inline emanifest-py/README.md:45]: Remove extra space after "authenticate" in line 45

Comment (dpgraham4401) [reply emanifest-py/README.md:45]: complete

File: emanifest-py/src/emanifest/client.py [hunk 2]
```diff
@@ -74,32 +73,68 @@ def decode(self):
 
 
 class RcrainfoClient(Session):
+    r"""
```
[the remark this answers is not in the document]
Comment (dpgraham4401) [reply emanifest-py/src/emanifest/client.py:76]: complete

File: emanifest-py/README.md [hunk 3]
```diff
@@ -14,140 +14,218 @@ refactoring.
...
+# For endpoints that return multipart/mixed bodies, you can access JSON with the resp.json()
+downloaded_json = resp.json()
+# The .zip file can be accessed in the resp.zip property.
+downloaded_attachment = resp.zip
 ```
 
-Once you've confirmed this is the correct site, you might search for manifests in transit from that site:
+### Regulators
+
+Starting with version > 3.0 emanifest python package, regulator can use the same methods as industry but with
```
[the remark this answers is not in the document]
Comment (dpgraham4401) [reply emanifest-py/README.md:156]: complete

File: emanifest-py/README.md [hunk 4]
```diff
@@ -14,140 +14,218 @@ refactoring.
...
 
 ### Methods
 
-After authenticating, you are ready to use the full functionality of the emanifest package. An introductory example
-script can be found [here](src/example.py). The RcrainfoClient class exposes a method for each API endpoint in one of
-the 10 service categories (the [methods can be viewed here](/emanifest-py/src/emanifest/client.py)). For more
-information about these services, visit the Swagger page of your selected
-environment. ([PREPROD](https://rcrainfopreprod.epa.gov/rcrainfo/secured/swagger/), [PROD](https://rcrainfo.epa.gov/rcrainfoprod/secured/swagger/)).
-API endpoints designed for use by other groups, such as regulators or industry users, will return 'Access Denied' errors
-if you are not authorized to access these resources in RCRAInfo.
```
[the remark this answers is not in the document]
Comment (dpgraham4401) [reply emanifest-py/README.md:85]: added back in

File: emanifest-py/README.md [hunk 5]
```diff
@@ -14,140 +14,218 @@ refactoring.
...
+Do not test using RCRAInfo (Production). To register for a testing
+account in preproduction, visit the [preprod site](https://rcrainfopreprod.epa.gov/rcrainfo/action/secured/login).
 
-If you are uncertain how to use a function, run help(em.FunctionName) in your Python environment. This will return a
-description of the function, any required inputs, and the formats of those inputs. For a list of all the functions
-contained in **emanifest** and additional information about this package, run help(emanifest) in your Python
-environment.
+Please note, the API of emanifest python package was substantially modified from version 2.0 to 3.0. We intend for the
+3.0 package version to be much more stable than its predecessors. We will continue to adhere to semantic versioning,
+and will not break backwards incompatibility within a major release.
```
[the remark this answers is not in the document]
Comment (dpgraham4401) [reply emanifest-py/README.md:224]: oops! haha thanks

File: emanifest-py/README.md [hunk 6]
```diff
@@ -14,140 +14,218 @@ refactoring.
...
+If you're ever in need of a methods signature and arguments, from the Python console, you can run
+
+```jupyterpython
+from emanifest import RcrainfoClient
+
+help(RcrainfoClient.get_bill)
+
+```
+
+to get a description of the method, any required inputs, and the formats of those inputs. For a list of all the
```
[the remark this answers is not in the document]
Comment (dpgraham4401) [reply emanifest-py/README.md:213]: fixed

**Pull request merged**